src/renate/benchmark/scenarios.py

```diff
@@ -244,39 +244,79 @@ def __init__(
 
 
 class IIDScenario(Scenario):
-    """A scenario splitting datasets into random equally-sized chunks.
+    """A scenario splitting datasets into random equally-sized chunks."""
+
+    def setup(self) -> None:
+        """Make assignments: val/train/test splits."""
+        self._data_module.setup()
+        proportions = [1 / self._num_tasks for _ in range(self._num_tasks)]
+        self._train_data = randomly_split_data(
+            self._data_module.train_data(), proportions, self._seed
+        )[self._chunk_id]
+        val_data = self._data_module.val_data()
+        if val_data:
+            self._val_data = randomly_split_data(val_data, proportions, self._seed)[self._chunk_id]
+        self._test_data = [self._data_module.test_data() for _ in range(self._num_tasks)]
+
+
+class _SortingScenario(Scenario):
+    """A scenario that _softly_ sorts a dataset by some score.
+
+    Randomness in the sorted order is induced by swapping the position of random pairs.
 
     Args:
         data_module: The source RenateDataModule for the the user data.
         num_tasks: The total number of expected tasks for experimentation.
+        randomness: A value between 0 and 1. 0 means no random swapping, 1 means random ordering.
```

Comment:
Maybe be more specific in describing what this parameter does, which is introducing a certain percentage of random swaps. I don't think it is accurate to say that setting it to `1` results in a (completely) random ordering.